src/main/java/io/tja/osrs/shattered_relics_fragment_presets/ShatteredRelicsFragmentPresetsPlugin.java

```diff
@@ -289,6 +293,32 @@ public MouseEvent mousePressed(MouseEvent mouseEvent) {
         return mouseEvent;
     }
 
+    private void swapFragmentsMenuEntry() {
+        /* This will only be called when the fragment window is open, we can do some simple filtering to ensure
+            we have the correct menu */
```

Comment:
Technically it could filter the menu names with `getTarget()` and then compare the target to a list of all fragment names, either statically defined or built in `onClientTick` where fragment comparisons are done -- but this should be a valid solution with less overhead.

Reply:
yeah, this is good enough imo